account_payment_order/views/account_payment_line.xml

```diff
@@ -17,7 +17,10 @@
                             domain="[('reconciled','=', False), ('account_id.reconcile', '=', True)] "
                         />
                         <!-- we removed the filter on amount_to_pay, because we want to be able to select refunds -->
-                        <field name="date" />
+                        <field
+                            name="date"
+                            attrs="{'readonly': [('ml_maturity_date', '!=', False)]}"
```

Comment:
I think the readonly condition may not be appropriated. I think it should be readonly in any state different from confirmed. In confirmed state, you should have the option to change it.

Reply:
In the 'Confirm Payments' process (`draft2open`) this date is re-defined https://github.com/OCA/bank-payment/pull/1296, so I think allowing to edit it would be a mistake (plus there would be discrepancy between transactions and payments that have already used that date).

Reply:
No, we precisely allow to change it in confirmed state, if you don't agree with the proposed one. On draft we avoid the confusion, and in the rest of the states is too late, as the file is already generated. That's why I say readonly when state != 'confirmed'.

Reply:
Checking this again for v16 I see that it is incorrect: the state condition is incorrect (it should be `open`) but also the transactions (`account.payment.line`) of a payment order has `readonly=True, states={"draft": [("readonly", False)]}`, so what should be the correct condition? (or maybe just `readonly=true`)

Reply:
Didn't you check the value of the state? I told you blindly in pseudo-code. Let's keep it readonly always then, as we don't need to change it. It was just for giving the option.